WordPress/src/main/java/org/wordpress/android/viewmodel/storage/StorageUtilsViewModel.kt

```diff
@@ -0,0 +1,37 @@
+package org.wordpress.android.viewmodel.storage
+
+import androidx.fragment.app.FragmentManager
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.delay
+import org.wordpress.android.modules.BG_THREAD
+import org.wordpress.android.util.StorageUtilsProvider
+import org.wordpress.android.util.StorageUtilsProvider.Source
+import org.wordpress.android.viewmodel.Event
+import org.wordpress.android.viewmodel.ScopedViewModel
+import javax.inject.Inject
+import javax.inject.Named
+
+class StorageUtilsViewModel @Inject constructor(
+    @Named(BG_THREAD) private val bgDispatcher: CoroutineDispatcher,
+    private val storageUtilsProvider: StorageUtilsProvider
+) : ScopedViewModel(bgDispatcher) {
+    private val _checkStorageWarning = MutableLiveData<Event<Unit>>()
+    val checkStorageWarning: LiveData<Event<Unit>> = _checkStorageWarning
+
+    fun onStorageWarningInit() {
+        launch(bgDispatcher) {
+            delay(DIALOG_CHECK_DELAY)
```

Comment:
I don't really understand why we need to add a delay here - is it because the observers may not be attached before? This relationship seems a bit flaky, maybe there's a way in which we can just trigger the action wrapped in `onStorageWarningCheck()` directly from EditPostActivity, without calling `onStorageWarningInit()` only to wait for the observer to be called back later on?

I'd go with something simpler like:
1. removing `onStorageWarningInit()` and its call
2. replace the call to `onStorageWarningInit()` with `onStorageWarningCheck()` in `EditPostActivity` 
3. remove the observer in EditPostActivity

I may be missing something, is there a reason not to go this way? Let me know 🙏

Reply:
I should have mentioned it in the description, my bad sir 🙃

The reason why I placed a delay there is to try to avoid visual glitches like this below with the keyboard on editor opening that can happen by timing races (maybe not the best approach but should do its job hopefully 😄  )

<p align=center><image src="https://user-images.githubusercontent.com/47797566/118638533-3b8ed480-b7d7-11eb-9ae9-66b73a232acb.gif" width=300/></p>

I preferred to post to a live data due to its nature of being lifecycle aware (so no issues in case the user could open the editor and exit soon after it). Considered something like Handler.postDelayed, but as said the live data approach seemed cleaner to me.

This said, I think I can do a better job with naming and used the `start()` approach we already use with VMs all over the places (also placed a guard on `savedInstanceState` to avoid showing the dialog including when the VM could be re-created when the activity is destroyed in background (see the `don't keep activities` forced scenario). 

Hope this works better now 🙇

Reply:
Oh I see! Understood, it's much clearer now! What do you think about adding this clarification in a comment in code? I'm still finding it difficult to understand if you read the code but lack this piece of context. Maybe also add a reference to this issue.

Nice call calling it `start()` and adding the safeguard for `savedInstanceState` 👍